fabric/src/main/java/org/samo_lego/taterzens/fabric/event/BlockInteractEventImpl.java

```diff
@@ -25,6 +25,11 @@ public BlockInteractEventImpl() {
      */
     @Override
     public InteractionResult interact(Player player, Level world, InteractionHand hand, BlockHitResult blockHitResult) {
-        return BlockEvent.onBlockInteract(player, world, blockHitResult.getBlockPos());
+        // check in if clause prevents crash (check for server side) on clients
+        // and prevents executing event handler twice (check for main hand)
```

Comment:
Just minor clarification: as player's npc will be null on client, it won't crash (at least it shouldn't :) ).

However, it will still double trigger (& it's clearer to check the side than relying on nulls).